File: AppSets-iOS/AppSets/com/app/appsets/util/http/DesignResponse.swift
//
//  DesignResponse.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/8.
//

import Foundation
import Hummingbird

class DesignResponse<D>: ResponseCodable where D : Codable {
    var code: Int

    var info: String?

    var data: D?
    
    init(code: Int, info: String? = nil, data: D? = nil) {
        self.code = code
        self.info = info
        self.data = data
    }
}

struct DesignResponseStatic {
    public static func notFound<D:Codable>()->DesignResponse<D>{
        return DesignResponse<D>(code: 404, data: nil)
    }
    public static func badRequest<D:Codable>()->DesignResponse<D>{
        return DesignResponse<D>(code: 400, data: nil)
    }
}

File: AppSets-Android/starter/src/main/java/xcj/app/starter/foundation/http/DesignResponse.kt
package xcj.app.starter.foundation.http

data class DesignResponse<A>(
    val code: Int = 0,
    val info: String? = null,
    val data: A? = null
) {
    val success: Boolean
        get() = code == 0

    companion object {
        val NO_DATA: DesignResponse<*>
            get() = DesignResponse(code = 0, data = null)
        val NOT_FOUND: DesignResponse<*>
            get() = DesignResponse(code = 404, data = null)
        val BAD_REQUEST: DesignResponse<*>
            get() = DesignResponse(code = 400, data = null)
    }
}